api/filter_dimensions.go

```diff
@@ -88,9 +90,25 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := api.isValidDatasetDimensions(ctx, v, []model.Dimension{req.Dimension}, filter.PopulationType); err != nil {
-		api.respond.Error(ctx, w, statusCode(err), err)
-		return
+	toValidate := []model.Dimension{req.Dimension}
+
+	if filter.Type == flexible {
+		err := api.isValidDatasetDimensions(ctx, v, toValidate, filter.PopulationType)
+		if err != nil {
+			api.respond.Error(ctx, w, statusCode(err), err)
+		}
+
+		finalDim = hydrateDimensions(toValidate, v.Dimensions)[0]
+
+	} else if filter.Type == multivariate {
+
+		multivariateDim, err := api.isValidMultivariateDimensions(ctx, toValidate, filter.PopulationType)
+		if err != nil {
+			api.respond.Error(ctx, w, statusCode(err), err)
+		}
+
+		finalDim = multivariateDim[0]
+
 	}
```

Comment:
Probably encapsulate this into method that can be reused on this `api` package, both on the `api/filter.go` and on the `api/filter_dimensions.go` files.

Reply:
Hello Paulo, please find encapsulation in the latest commit, alongside ammended tests to marry up the check with expectations. I will leave  you to resolve the comment